### PR TITLE
store/copr: polish the tiflash-tikv fallback function.

### PR DESCRIPTION
**Pull request by hanfei1991**
### What problem does this PR solve?

Problem Summary:
fallback to tikv when batch cop or mpp meets io error

### What is changed and how it works?

What's Changed:
Convert all kinds of io error in BatchCop and MPP to "TiFlashTimeout" error which can be identified by fallback function and retry in TiKV mode.
For now, mpp io error cannot be retried by itself. So even not a fatal error can be routed to tikv's retry. This problem can be tackled after we support mpp client's retry.

### Related changes

- PR to update `unistore`: https://github.com/ngaut/unistore/pull/438 fix a leak in unistore after processing mpp request.

### Check List 

Tests 

- Unit test for batch cop, mpp test.

### Release note 

- store/copr: polish the tiflash-tikv fallback function.

### CODE REVIEW

File: store/mockstore/unistore/cophandler/mpp_exec.go [hunk 1]
```diff
@@ -231,6 +236,12 @@ func (e *exchRecvExec) open() error {
 }
 
 func (e *exchRecvExec) next() (*chunk.Chunk, error) {
+	if !e.inited {
```
Comment (winoros) [inline store/mockstore/unistore/cophandler/mpp_exec.go:239]: Why move this part?

Comment (hanfei1991) [reply store/mockstore/unistore/cophandler/mpp_exec.go:239]: I'm trying to pass all the errors (including injected error) to ErrChannel, which can not be implemented in the "open" function. The error is returned by "next" function and is captured in the "exchSender", then be passed to the next task, util the error approches client.

**Comment by winoros** [comment]
but both of us have no permission to approve, i'm asking others for help.

File: server/conn_test.go [hunk 2]
```diff
@@ -723,37 +747,48 @@ func (ts *ConnTestSuite) TestFallbackToTiKVWhenTiFlashIsDown(c *C) {
 	tk := testkit.NewTestKitWithInit(c, ts.store)
 	cc.ctx = &TiDBContext{Session: tk.Se, stmts: make(map[int]*TiDBStatement)}
 
-	tk.MustExec("set @@session.tidb_enable_tiflash_fallback_tikv = 1")
+	tk.MustExec("use test")
```
Comment (xuyifangreeneyes) [inline server/conn_test.go:750]: Executing "use test" may be unnecessary since `testkit.NewTestKitWithInit` already includes it.

**Review by lysu — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Pull request merged**